src/lib/transaction/Transaction.svelte

```diff
@@ -17,33 +17,42 @@
         const keyPair = await getStellarKeypair(privateKey);
         return keyPair;
     }
-
     const keyPair = getKeyPair();
 
     let tx: Transaction;
     let operationComponents: typeof OperationComponentTypes[] = [];
-
     const isValidXdr = writable(false);
-    const xdrValue = location.search.substring(5);
+
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const xdrValue = urlParams.get('xdr')?.replace(/\s/g, '+');
+    const description = urlParams.get('description');
 
     try {
-        $isValidXdr = xdr.TransactionEnvelope.validateXDR(xdrValue, 'base64');
-        tx = new Transaction(xdrValue, import.meta.env.VITE_HORIZON_NETWORK_PASSPHRASE);
+        $isValidXdr = xdr.TransactionEnvelope.validateXDR(xdrValue!, 'base64');
+        tx = new Transaction(xdrValue!, import.meta.env.VITE_HORIZON_NETWORK_PASSPHRASE);
 
         const dynamicOperationComponentFactory = new DynamicOperationComponentFactory();
 
         for (let i = 0; i < tx.operations.length; i++) {
             let operationComponent = dynamicOperationComponentFactory.create(tx, tx.operations[i]!);
             operationComponents.push(operationComponent);
         }
-    } catch (error) {
-        console.error(error);
+    } catch (e) {
+        console.error({ invalidUrl: e });
```

Comment:
if you are passing this arbitrary object then instead, you need to create an interface for it so it can be reused somewhere else